Content.Client/_Impstation/DroneVision/DroneVisionSystem.cs

```diff
@@ -0,0 +1,61 @@
+using Content.Shared._Impstation.Overlays;
+using Content.Shared._DV.CCVars;
+using Robust.Client.Graphics;
+using Robust.Shared.Configuration;
+using Robust.Shared.Player;
+
+namespace Content.Client._Impstation.Overlays;
+
+public sealed partial class DroneVisionSystem : EntitySystem
+{
+    [Dependency] private readonly IOverlayManager _overlayMan = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly ISharedPlayerManager _playerMan = default!;
+
+    private DroneVisionOverlay _overlay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<DroneVisionComponent, ComponentInit>(OnDroneVisionInit);
+        SubscribeLocalEvent<DroneVisionComponent, ComponentShutdown>(OnDroneVisionShutdown);
+        SubscribeLocalEvent<DroneVisionComponent, LocalPlayerAttachedEvent>(OnPlayerAttached);
+        SubscribeLocalEvent<DroneVisionComponent, LocalPlayerDetachedEvent>(OnPlayerDetached);
+
+        Subs.CVar(_cfg, DCCVars.NoVisionFilters, OnNoVisionFiltersChanged);
+
+        _overlay = new();
+    }
+
+    private void OnDroneVisionInit(EntityUid uid, DroneVisionComponent component, ComponentInit args)
+    {
+        if (uid == _playerMan.LocalEntity && !_cfg.GetCVar(DCCVars.NoVisionFilters))
```

Comment:
given the mechanical backing, it might be worth forcing this even with filters disabled in options